### ISSUE TITLE
Fix Request: false positive in AvoidFutureJoinWithoutTimeout

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#198]
false positive of String.join in a streams:
```
String extract(String items) {
 List<String> pairs = Arrays.stream(items.split(","))
 .map(item -> item.trim().split("="))
 .filter(itemPair -> "ID".equalsIgnoreCase(itemPair[0]))
 .map(itemPair -> String.join(":", itemPair)) // false positive
 .collect(Collectors.toList());
 return String.join(" --- ", pairs);
}
```
